spec/oplogjam/update_spec.rb

```diff
@@ -248,7 +248,21 @@ module Oplogjam
         update = build_update(1, '$set' => { 'a.1' => 1 })
         update.apply('foo.bar' => table)
 
-        expect(table.first).to include(document: Sequel.pg_jsonb('_id' => 1, 'a' => [nil, 1]))
+        expect(table.first).to include(document: Sequel.pg_jsonb('_id' => 1, 'a' => [1]))
+      end
+
+      it 'applies non-zero based array update to {"a"=>[]}' do
```

Comment:
I think is is worth adding more test cases, for documentation purposes. I'm thinking in the case where `a` is a non-empty array.

Reply:
Will do 👍

Reply:
Check 'em out: https://github.com/altmetric/oplogjam/pull/1/commits/58a619b71d90505464306f2b726663ad9d8fd9c7

Reply:
Here's the commit with added 'gaps-removal' test-case https://github.com/altmetric/oplogjam/pull/1/commits/ead17dd59b56eb2093f581575160372632ea9200